### PR TITLE
Multi cluster doc improve

### PR DESCRIPTION
**Pull request by zijun726911**
**What type of PR is this?** documentation

**Which issue does this PR fix**: Add new multi cluster doc and diagram to reflect the new controller behaviour, deprecate the outdated description

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by mikhail-aws — CHANGES_REQUESTED** [review]
Need to make it crisp(er). Documents describes "How" to setup, but does not explain "Why" it's recommended way.

Picture is complicated and mix of different ideas. Different names - "my-hotel", "my-httproute", "service". My-hotel looks foreign (or others?). We should have it clear what are k8s and aws/lattice concepts in picture. May be dont mix them together, if you want them on the same picture try to use layers.

**Comment by zijun726911** [comment]
I can change "my-hotel" to "my-gateway". The old doc didn't mention why it's recommended. I can add few sentences in the new doc to explain it.

> We should have it clear what are k8s and aws/lattice concepts in picture. May be dont mix them together, if you want them on the same picture try to use layers.
I think only lattice resource need to be presented is service network since its no longer 1:1 map to the k8s gateway. for other k8s resource I can use k8s icon and word "k8s" to make it clear. Not quite sure how the layers looks like

**Review by mikhail-aws — APPROVED** [review]

File: docs/guides/multi-cluster.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  - Multiple application workload(s) (Pods, Deployments etc.)
+  - Multiple `Service(s)` for application workload(s)
+  - Multiple `ServiceExport(s)`, that export kubernetes application Service(s) to the "config cluster"
+- One extra dedicated "config cluster", which is act as a "service mesh control plane" and it should include following kubernetes objects:
+  - One `Gateway` that has __same name__ as the manually created VPC Lattice service network name
+  - Multiple `ServiceImport(s)`, that reference to kubernetes application services that export from workload cluster(s)
+  - Multiple `HTTPRoute(s)`,`GRPCRoute(s)`, that have rules backendRefs to `ServiceImport(s)` that referring kubernetes application service(s) in workload cluster(s)
+
+
+You can see this similar production use case at Airbnb: [airbnb mullti-cluster setup](https://www.youtube.com/watch?v=1D8lg36ZNHs)
```
Comment (solmonk) [inline docs/guides/multi-cluster.md:18]: I think we should remove this, instead we should have good reasons for this recommendation

File: docs/guides/multi-cluster.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Here is a recommended multi-cluster architecture if you'd like to setup kubernetes service-to-service communications across multiple clusters.
+
+Suppose your organization would like to have one "service mesh" that spans many clusters in one aws account, this service mesh should include the following components:
+- One manually created VPC Lattice service network, (it could create by either AWS Console, CLI, CloudFormation, Terraform or any other tools)
+- Create `VpcServiceNetworkAssociations` between VPC Lattice service network and each config cluster's VPC and workload clusters' VPCs
+- Multiple workload cluster(s), that are used to run application workload(s). workload cluster(s) should only have following workloads related kubernetes objects:
+  - Multiple application workload(s) (Pods, Deployments etc.)
+  - Multiple `Service(s)` for application workload(s)
+  - Multiple `ServiceExport(s)`, that export kubernetes application Service(s) to the "config cluster"
```
Comment (solmonk) [inline docs/guides/multi-cluster.md:11]: I don't think we need `Multiple` on these three bullet points

File: docs/guides/multi-cluster.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+- One extra dedicated "config cluster", which is act as a "service mesh control plane" and it should include following kubernetes objects:
+  - One `Gateway` that has __same name__ as the manually created VPC Lattice service network name
+  - Multiple `ServiceImport(s)`, that reference to kubernetes application services that export from workload cluster(s)
+  - Multiple `HTTPRoute(s)`,`GRPCRoute(s)`, that have rules backendRefs to `ServiceImport(s)` that referring kubernetes application service(s) in workload cluster(s)
```
Comment (solmonk) [inline docs/guides/multi-cluster.md:15]: Same, I don't think we need `Multiple` on these bullet points

File: docs/guides/multi-cluster.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Recommended Multi-Cluster Architecture
+
+Here is a recommended multi-cluster architecture if you'd like to setup kubernetes service-to-service communications across multiple clusters.
+
+Suppose your organization would like to have one "service mesh" that spans many clusters in one aws account, this service mesh should include the following components:
```
Comment (solmonk) [inline docs/guides/multi-cluster.md:5]: * I feel like we should avoid using the term "service mesh" here, as it is not a VPC Lattice terminology. saying something like one large infrastructure will do the job.
* aws -> AWS

**Review by solmonk — CHANGES_REQUESTED** [review]

File: docs/guides/multi-cluster.md (hunk 1 above)
Comment (zijun726911) [reply docs/guides/multi-cluster.md:18]: Any idea on what good reasons we could say?

File: docs/guides/multi-cluster.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+Here is a recommended multi-cluster architecture if you'd like to setup kubernetes service-to-service communications across multiple clusters.
+
+Suppose your organization would like to have one large infrastructure that have many clusters in one AWS account. It's recommended your infrastructure includes the following components:
+- One manually created VPC Lattice service network, (it could create by either AWS Console, CLI, CloudFormation, Terraform or any other tools)
+- Create `VpcServiceNetworkAssociations` between VPC Lattice service network and each config cluster's VPC and workload clusters' VPCs
+- Multiple workload cluster(s), that are used to run application workload(s). Workload cluster(s) should only have following workloads related kubernetes objects:
+  - Application workload(s) (`Pod(s)`, `Deployment(s)` etc.)
+  - `Service(s)` for application workload(s)
+  - `ServiceExport(s)`, that export kubernetes application Service(s) to the "config cluster"
+- One extra dedicated "config cluster", which is act as a "service network control plane" and it should include following kubernetes objects:
```
Comment (solmonk) [inline docs/guides/multi-cluster.md:12]: is act -> acts

File: docs/guides/multi-cluster.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
+# Recommended Multi-Cluster Architecture
+
+Here is a recommended multi-cluster architecture if you'd like to setup kubernetes service-to-service communications across multiple clusters.
+
+Suppose your organization would like to have one large infrastructure that have many clusters in one AWS account. It's recommended your infrastructure includes the following components:
+- One manually created VPC Lattice service network, (it could create by either AWS Console, CLI, CloudFormation, Terraform or any other tools)
```
Comment (solmonk) [inline docs/guides/multi-cluster.md:6]: create -> be created

File: docs/guides/multi-cluster.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+- Create `VpcServiceNetworkAssociations` between VPC Lattice service network and each config cluster's VPC and workload clusters' VPCs
+- Multiple workload cluster(s), that are used to run application workload(s). Workload cluster(s) should only have following workloads related kubernetes objects:
+  - Application workload(s) (`Pod(s)`, `Deployment(s)` etc.)
+  - `Service(s)` for application workload(s)
+  - `ServiceExport(s)`, that export kubernetes application Service(s) to the "config cluster"
+- One extra dedicated "config cluster", which is act as a "service network control plane" and it should include following kubernetes objects:
+  - One `Gateway` that has __same name__ as the manually created VPC Lattice service network name
+  - `ServiceImport(s)`, that reference to kubernetes application services that export from workload cluster(s)
```
Comment (solmonk) [inline docs/guides/multi-cluster.md:14]: export -> are exported

File: docs/guides/multi-cluster.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  - `HTTPRoute(s)`,`GRPCRoute(s)`, that have rules backendRefs to `ServiceImport(s)` that referring kubernetes application service(s) in workload cluster(s)
+
+![config cluster and multiple workload clusters](../images/multi-cluster.png)
+
+
+Above architecture is recommended for following reasons:
+- Putting all Route(s) and only one Gateway in the config cluster is easier to manage and have less chance to have conflict.
+- You are easier to have seperated roles in your organization. The infra operator team only manage the resource in the config cluster and VPC Lattice resource, and the application developer team only manage the resource in the workload cluster(s).
```
Comment (solmonk) [inline docs/guides/multi-cluster.md:22]: you are -> it is

File: docs/guides/multi-cluster.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+
+
+Following steps show you how to set up this recommended multi-cluster architecture with 1 config cluster and 2 workload clusters.
+1. Create 3 k8s clusters: `config cluster`, `workload cluster-1`, `workload cluster-2`. Install AWS gateway api controller in each cluster, you could follow this instruction [deploy.md](deploy.md)
```
Comment (solmonk) [inline docs/guides/multi-cluster.md:27]: AWS gateway api controller -> AWS Gateway API Controller

File: docs/guides/multi-cluster.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    ```
+1. Setup following resource in the config cluster:
+    ```
+    kubectl apply -f examples/my-gateway.yaml
+    kubectl apply -f examples/my-httproute.yaml
+    kubectl apply -f examples/service-1-import.yaml
+    kubectl apply -f examples/service-2-import.yaml
+    ```
+1. At this point, the connectivity setup finished, pods in workload cluster1 are able to communicate with `service-2` in workload cluster2 (and vice versa) via the `my-httproute` dns name.
+1. Furthermore, you could have more workloads clusters to join the `my-gateway` service network by creating the `ServiceNewtorkAssociation(s)`, they will all be able to communicate with `service-1` and `service-2` via the `my-httproute` dns name and path matching.
```
Comment (solmonk) [inline docs/guides/multi-cluster.md:48]: dns -> DNS

File: docs/guides/multi-cluster.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
+# Recommended Multi-Cluster Architecture
+
+Here is a recommended multi-cluster architecture if you'd like to setup kubernetes service-to-service communications across multiple clusters.
+
+Suppose your organization would like to have one large infrastructure that have many clusters in one AWS account. It's recommended your infrastructure includes the following components:
+- One manually created VPC Lattice service network, (it could create by either AWS Console, CLI, CloudFormation, Terraform or any other tools)
+- Create `VpcServiceNetworkAssociations` between VPC Lattice service network and each config cluster's VPC and workload clusters' VPCs
```
Comment (solmonk) [inline docs/guides/multi-cluster.md:7]: should be `VpcAssociationPolicy` or just plain vpc service network association, this sounds like it is a resource that we can create through k8s

**Review by solmonk — COMMENTED** [review]
I think the contents are fine, we can start from this. Just found a few expressions that could be better

**Review by solmonk — APPROVED** [review]

**Pull request merged**